### DISCUSSION TITLE
Running Unix executable inside of Electron app

### DISCUSSION THREAD

**Discussion by rayisooo** [atom/atom#22574] [Q&A]
im trying to run an executable with node js from an electron application but its not working or doing anything ive checked multiple times that my code is correct im using spawn childprocess the application was a python script I turned into an executable using PyInstaller it works when I click it from my finder but when electron node js tries to call it and pass arguments to it using child process it does nothing it doesn't give me any logs or any errors.

here's the code 

 `

    const {spawn} = require('child_process');
    const child = spawn(__dirname +'/bots/supreme_encrypted',my_data);

    child.stdout.on('data',(data)=>{
      console.log(`STD OUT : ${data}`);
      status_text.innerHTML =  `${data}`; //updates text in status td
    });
    child.stderr.on('data',(data)=>{
      console.log(`std ERR  : ${data}`);
      status_text.innerHTML =  `${data}`;
    });
    child.on('error',(error)=>{
      console.log(`error : ${error.message}`)
    });

    child.on('exit',(code,signal)=>{
      if(code){console.log(`Process exit : ${code}`)
      status_text.innerHTML =  `${code},${signal}`;
    };
      if(signal){
        console.log(`Process killed with signal ${signal}`)
        status_text.innerHTML =  `${signal}`;
     };
    });`

any idea why its not running?

**Comment by icecream17** [comment]
Maybe your child process isn't doing anything or is stuck in an infinite loop? That way none of the handlers are called

Maybe try a global `console.log` call and see if that does something

Also there's a backtick at the end of your code
Also you can add syntax highlighting to code by adding the language next to the backticks: ` ```js ` does javascript syntax highlighting

`````js

// Example of syntax highlighting

```js
This code is syntax highlighted in Javascript
```

purpleFunction()
`````
